infrastructure/code/backend_service.go

```diff
@@ -349,3 +351,110 @@ func (s *SnykCodeHTTPClient) checkResponseCode(r *http.Response) error {
+		return nil, AutofixStatus{}, err
+	}
+	log.Debug().Str("method", method).Str("requestId", requestId).Msg("API: Retrieving analysis for bundle")
+	defer log.Debug().Str("method", method).Str("requestId", requestId).Msg("API: Retrieving analysis done")
+
+	requestBody, err := s.autofixRequestBody(&options)
+	if err != nil {
+		log.Err(err).Str("method", method).Str("requestBody", string(requestBody)).Msg("error creating request body")
+		return nil, AutofixStatus{}, err
+	}
+
+	responseBody, err := s.doCall(span.Context(), "POST", "/autofix/suggestions", requestBody)
+	failed := AutofixStatus{message: "FAILED"}
+	if err != nil {
+		log.Err(err).Str("method", method).Str("responseBody", string(responseBody)).Msg("error response from analysis")
+		return nil, failed, err
+	}
+
+	var response AutofixResponse
+	err = json.Unmarshal(responseBody, &response)
+	if err != nil {
+		log.Err(err).Str("method", method).Str("responseBody", string(responseBody)).Msg("error unmarshalling")
+		return nil, failed, err
+	}
+
+	log.Debug().Str("method", method).Str("requestId", requestId).Msgf("Status: %s", response.Status)
+
+	if response.Status == failed.message {
+		log.Err(err).Str("method", method).Str("responseStatus", response.Status).Msg("analysis failed")
+		return nil, failed, SnykAutofixFailedError{Msg: string(responseBody)}
+	}
+
+	if response.Status == "" {
+		log.Err(err).Str("method", method).Str("responseStatus", response.Status).Msg("unknown response status (empty)")
+		return nil, failed, SnykAutofixFailedError{Msg: string(responseBody)}
+	}
+	status := AutofixStatus{message: response.Status}
+	if response.Status != completeStatus {
+		return nil, status, nil
+	}
+
+	suggestions := response.toAutofixSuggestions(options.filePath)
+	return suggestions, AutofixStatus{message: response.Status}, nil
+}
+
+func (s *SnykCodeHTTPClient) autofixRequestBody(options *AutofixOptions) ([]byte, error) {
+	orgName := "unknown"
+	if config.CurrentConfig().GetOrganization() != "" {
+		orgName = config.CurrentConfig().GetOrganization()
+	}
+
+	issueData, ok := options.issue.AdditionalData.(IssueData)
+	if !ok {
+		return nil, SnykAutofixFailedError{Msg: "Issue does not contain IssueData"}
+	}
+	// TODO(alex.gronskiy): see RFC. The `issue.AdditionalData.RuleID` has a form of `<lang>/<ruleID>` so we need to
+	// decide how to use the language here. Currently, we only send `<ruleID>` part, ignoring the
+	// language.
+	ruleIdSplit := strings.Split(issueData.RuleId, "/")
+	if len(ruleIdSplit) != 2 {
+		return nil, SnykAutofixFailedError{Msg: "Issue's ruleID does not follow <lang>/<ruleKey> format"}
+	}
+
+	request := AutofixRequest{
+		Key: AutofixRequestKey{
+			Type:     "file",
+			Hash:     options.bundleHash,
+			FilePath: options.filePath,
+			RuleId:   ruleIdSplit[1],
+			LineNum:  options.issue.Range.Start.Line + 1,
+		},
+		AutofixContext: AutofixContext{
+			Initiatior: "IDE",
```

Comment:
The context here is the same as the `AnalysisContext` for `AnalysisRequest` (see `analysisRequestBody`). We can extract this to a func.

Reply:
✅